tests/main/interfaces-network-manager/task.yaml

```diff
@@ -0,0 +1,53 @@
+summary: Ensure that the network-manager interface works
+
+description: |
+    The network-manager interface gives privileged access to configure and
+    observe networking.
+    The test uses a snap which plugs the network manager interface. Then it is
+    validated that the snap autoconects and can create a new connection.
+    Connection against network devices cannot be validated on a virtual machine
+    due to network-manager being configured not to managed them.
+
+# run only against the amd64 VM, we cannot run this on arm/arm64
+# boards because the (wifi) network is already managed by netplan
+# there and when n-m gets installed/removed it will hang when
+# trying to deconfigure the wifi network which is already owned.
+systems: [ubuntu-core-16-64, ubuntu-core-16-32]
+
+execute: |
+    CONNECTED_PATTERN="^network-manager:service +network-manager:nmcli"
+    DISCONNECTED_PATTERN="^network-manager:service +-$"
+
+    echo "Give a network-manager snap is installed"
+    snap install network-manager
+
+    # using wait_for_service is not enough, systemd considers the service
+    # active even when it is not (yet) listening to dbus
+    for i in $(seq 300); do
```

Comment:
Would it make sense to move this into a function? I see an existing regression/nmcli test using the same 6-line workaround, both could benefit from calling a simple helper function.

Reply:
Well, in fact I am gonna remove the regression test due to this on includes the other one and add more checks. Thanks for the observation.